lib/fastlane/plugin/revenuecat_internal/helper/versioning_helper.rb

```diff
@@ -161,18 +161,91 @@ def self.detect_bump_type(version_name_a, version_name_b)
         end
       end
 
+      def self.validate_input_if_appending_phc_version?(append_hybrid_common_version, include_prereleases, hybrid_common_version)
+        if append_hybrid_common_version
+          UI.user_error!("Appending the PHC version to prerelease versions violates SemVer.") if include_prereleases
+          UI.user_error!("Cannot append a nil PHC version.") if hybrid_common_version.nil?
+          UI.user_error!("Cannot append a blank PHC version.") if hybrid_common_version.strip.empty?
+        end
+      end
+
+      def self.validate_new_version_if_appending_phc_version?(append_hybrid_common_version, new_version_number, hybrid_common_version)
+        if append_hybrid_common_version
+          if new_version_number.include?("-")
+            UI.user_error!("Appending the PHC version to prerelease versions violates SemVer.")
+          end
+          if new_version_number.include?("+") && new_version_number.partition("+").last != hybrid_common_version
+            UI.user_error!(
+              "Asked to append PHC version (+#{hybrid_common_version}), " \
+              "but the provided version (#{new_version_number}) already has metadata " \
+              "(+#{new_version_number.partition('+').last})."
+            )
+          end
+        end
+      end
+
+      def self.append_phc_version_if_necessary(append_hybrid_common_version, include_prereleases, hybrid_common_version, new_version_number)
+        if should_ask_to_append_phc_version?(append_hybrid_common_version, include_prereleases, hybrid_common_version, new_version_number)
+          if UI.confirm("Would you like to append the PHC version (+#{hybrid_common_version})?")
+            return "#{new_version_number}+#{hybrid_common_version}"
+          end
+        elsif should_append_phc_version?(append_hybrid_common_version, include_prereleases, hybrid_common_version, new_version_number)
+          UI.important(
+            "Appending PHC version (+#{hybrid_common_version}) to new version (#{new_version_number}), as instructed."
+          )
+          return "#{new_version_number}+#{hybrid_common_version}"
+        end
+        return new_version_number
+      end
+
+      private_class_method def self.should_ask_to_append_phc_version?(append_hybrid_common_version, include_prereleases, hybrid_common_version, new_version_number)
+        UI.interactive? &&
+        append_hybrid_common_version.nil? &&
```

Comment:
should this check for `nil` or should it check the value of `append_hybrid_common_version`?

It feels like these two private methods `should_ask_to_append_phc_version` and `should_append_phc_version` could be combined into one and then check if the UI is interactive to show a `UI.confirm` or a `UI.important`. But maybe I am missing something

Reply:
It was checking for `nil`, because that meant no decision had been made about whether we should append the PHC version or not, and so we should ask the user.

However this logic is now removed. We never ask the user anymore. Repos that need this will have the relevant parameter hardcoded to true. See also this comment thread: https://github.com/RevenueCat/fastlane-plugin-revenuecat_internal/pull/67#discussion_r1700360264.